### PR TITLE
June14merge

### PR DESCRIPTION
**Pull request by gmartin102**
Cumulative merge of changes from Quicklogic

### CODE REVIEW

File: rtl/apb_i2cs/rtl/FIFO_sync_256x8.v [hunk 1]
```diff
@@ -0,0 +1,290 @@
+/* ----------------------------------------------------------------------------
```
Comment (zarubaf) [inline rtl/apb_i2cs/rtl/FIFO_sync_256x8.v:1]: We need a file header here :-)

File: rtl/apb_i2cs/rtl/RAM_256x8_behavioral.v [hunk 2]
```diff
@@ -0,0 +1,52 @@
+/* ----------------------------------------------------------------------------
```
Comment (zarubaf) [inline rtl/apb_i2cs/rtl/RAM_256x8_behavioral.v:1]: License header, please

File: rtl/apb_i2cs/rtl/apb_i2cs.v [hunk 3]
```diff
@@ -0,0 +1,198 @@
+/* ----------------------------------------------------------------------------
```
Comment (zarubaf) [inline rtl/apb_i2cs/rtl/apb_i2cs.v:1]: License header, please

File: rtl/apb_i2cs/rtl/apb_slv_interface.v [hunk 4]
```diff
@@ -0,0 +1,121 @@
+
```
Comment (zarubaf) [inline rtl/apb_i2cs/rtl/apb_slv_interface.v:1]: License header, please

File: rtl/apb_i2cs/rtl/i2c_peripheral_interface.v [hunk 5]
```diff
@@ -0,0 +1,457 @@
+/* ----------------------------------------------------------------------------
```
Comment (zarubaf) [inline rtl/apb_i2cs/rtl/i2c_peripheral_interface.v:1]: License header, please

File: rtl/apb_i2cs/rtl/i2c_peripheral_interface.v [hunk 6]
```diff
@@ -0,0 +1,457 @@
...
+  assign i2c_sda_o = sda_out;
+  assign i2c_reg_addr_o = reg_addr;
+  assign i2c_reg_wdata_o = in_byte;
+
+  assign i2c_reg_rd_byte_complete_o = i2c_rd_byte_complete;
+
+endmodule
+
+
+//////////////////////////////////////////////////////////////////////
```
Comment (zarubaf) [inline rtl/apb_i2cs/rtl/i2c_peripheral_interface.v:414]: Ah, I think that needs to go on top. But I see problems with the license, we generally can't have LGPL licensed things in here as this would taint the rest of the liberal licenses source code.

**Review by zarubaf — CHANGES_REQUESTED** [review]
Generally looks okay from my side except for the LGPL License of the I2C IP that we can't accept here (and would also be problematic in your project). Any idea what to do about this? CC/ @MikeOpenHWGroup @jeremybennett @davideschiavone 

Please also merge the latest OpenHW Group master - so that we do not introduce a lint regression ;-D

File: rtl/apb_i2cs/rtl/i2c_peripheral_interface.v (hunk 6 above)
Comment (gmartin102) [reply rtl/apb_i2cs/rtl/i2c_peripheral_interface.v:414]: Since this is QL source I think we can use the same license header as we use on the other QL source?  If so, I'll fix and resubmit.  --- I thought I was up to date with OpenhwGroup master, but I will also double check that.

**Comment by MikeOpenHWGroup** [comment]
Its my understanding that OpenHW cannot accept code covered by the LGPL License.  I will confirm.  @DBees, can you comment?

**Comment by davideschiavone** [comment]
> 
> 
> Its my understanding that OpenHW cannot accept code covered by the LGPL License. I will confirm. @DBees, can you comment?

My understanding is also that we can accept permissive licence only contributions, as for example Solderpad and Apache

**Comment by MikeOpenHWGroup** [comment]
Update on the LGPL license.  The OpenHW Group cannot accept the LGPL for repositories covered by the [CORE-V Eclipse Foundation project](https://projects.eclipse.org/projects/openhw.corev/developer).  Recall that the Eclipse Foundation process helps us ensure the correct licensing and attribution of our open-source artifacts.  However, core-v-mcu is _not_ covered by the CORE-V Eclipse Foundation project _at this time_.

So, this means:
- We can accept the LGPL licensing for this PR.
- If we do that, the LGPL licensing will need to be resolved before OpenHW can make this a CORE-V project with proper attribution of open source licensing.  This is done through a formal "CQ Process".

Bottom line: it would be far better to resolve this now by removing the LGPL license and adding a permissive licence to this contribution, as for example Solderpad and Apache.  This can only be done by the current copyright holder.

@gmartin102, can you change the copyright on the affected files and update this PR?

**Comment by gmartin102** [comment]
@zarubaf - I think I added the Solderpad license header to the new i2c IP, If there is something else in error please let me know what and where and I'll correct it.  I'm a little confused about the LGPL license and where is was referenced

**Comment by zarubaf** [comment]
The problem is that the I2C core seems to originate from OpenCores and was originally licensed under LGPL, at least that’s what the stray license header indicated. We can not change the license - that must be done by the original author.

**Review by gmartin102 — COMMENTED** [review]
@zarubaf or @davideschiavone -- Do i need to fix anything on my end? or can you resolve the  conflicts for this PR?

**Comment by davideschiavone** [comment]
> @zarubaf or @davideschiavone -- Do i need to fix anything on my end? or can you resolve the conflicts for this PR?

I don't have the rights to do it but I can help you if you need -

**Comment by gmartin102** [comment]
@davideschiavone  I don't have rights to resolve either, it seems two files are in conflict, the README.md -- which should be an easy resolution and the core-v-mcu.core file -- which the QL build process doesn't use.  we moved that file to the root directory as core-v-mcu-emul.core file to avoid the warnings(?) about referencinf something outside the current directory tree.  -- So bottom line I don't see these as issues for the build

**Comment by zarubaf** [comment]
@gmartin102 That should do it!
```
git remote add upstream git@github.com:openhwgroup/core-v-mcu.git
git fetch upstream
git merge upstream/master
```

**Comment by gmartin102** [comment]
@davideschiavone , @zarubaf , Little bit of research and I think i fixed the conflicts.  but I am seeing a lint checkk fail, not sure how to fix that.  please advise.

**Comment by davideschiavone** [comment]
the make lint seems to fail... what do you get in your terminal?

**Comment by gmartin102** [comment]
Davide, I am seeing the following message,
ERROR: openhwgroup.org<http://openhwgroup.org>:systems:core-v-mcu-orig:0 : Fileset 'verilator-waiver', requested by target 'default', was not found

The issue seems to arise from the fact the Tim has moved? Or copied the core-v-mcu.core file that was in the rtl/core-v-mcu directory to the top level directory and renamed it core-v-mcu-emulation in the top level and changed  some reference in the top level core-v-mcu.core file.
I believe he was seeing a warning about not having files pulled from directories not in the current tree.  Lots of rtl is in parallel directories to the rtl/core-v-mcu.
I see the Florian and Jeremy have been adding options for Verilator lint the rtl/core-v-mcu/core-v-mcu.core file and I suspect those changes need to be properly implemented in the new core-v-emulation.core.
I’ll investigate more and work with Tim to see if I can figure out how to fix this without losing Florian/Jeremys intent.

Greg



On Jul 2, 2021, at 9:27 AM, Pasquale Davide Schiavone ***@***.******@***.***>> wrote:

CAUTION: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe.



the make lint seems to fail... what do you get in your terminal?

**Comment by gmartin102** [comment]
@davideschiavone -- can you approve this pull request?  I think all issues are resolved.

**Review by zarubaf — APPROVED** [review]
Perfect, thanks @gmartin102!
